Automattic/vip-go-mu-plugins
# CI Observability: Integrate ForeSight to tests phases of CI

## Description

This PR makes few enhancements and additions to the Github workflows for test phases. 

1. Integrate ForeSight to the E2E test phases - This is being done to increase visibility of execution time and fragility at test level in the pipelines. 
2. Change Playwright installation to include only Chromium - The E2E tests are currently only being run on Chromium in the CI. Limiting the installation to single browser/engine in the workflow helps us reduce pipeline execution time
3. Playwright config enhancement - Simpler logic to keep the test worker count at 2 in CI and `default` in the local execution

## Changelog Description

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

The E2E test should run properly after this PR.

How to verify:
1. `npm run cypress:run` should correctly run the relevant E2Es
2. `cd __tests__/e2e`
3. `npm run test` should correctly run all the relevant E2Es
